### ISSUE TITLE
Feature Request: Expose Snippets Via JSON

### ISSUE THREAD

**Issue by vkhitrin** [lemoony/snipkit#202]
Hey @lemoony, I enjoy `snipkit` a lot since this is a great aggregator of snippets.

I was wondering if it is part of the scope of this project to have the ability to expose available snippets via JSON in addition to the TUI.
If we can access all the snippets as a JSON, we can leverage `snipkit` and use this info for various GUI/front ends.

For example, I would like to have an [Alfred](https://www.alfredapp.com) workflow or a [Raycast](https://www.raycast.com) extension to act as the front end.

**Comment by vkhitrin** [comment]
@lemoony, I am slowly trying to get familiar with the code base (as seen in #203).
If this is something you don't wish to work on, I will try my best to propose a draft in the future.

**Comment by lemoony** [comment]
@vkhitrin Sorry for the delayed replay, right now I'm drowning in work. I will look at the MR in the next couple of days! I really appreciate the contribution :)

**Comment by lemoony** [comment]
Ok, I think I understood. However, the JSON could potentially be quite large.

Is it enough for your use to dump all snippets managed by snipkit to stdout or do you want to pass some filter parameter?

**Comment by vkhitrin** [comment]
> Ok, I think I now understand your proposal. However, the JSON could potentially be quite large.

True, it all depends on the amount of snippets a user has. I don't have many snippets right now, so I was not too concerned about fetching all snippets.

> Is it enough for your use to dump all snippets managed by snipkit to stdout or do you want to pass some filter parameter?

This is a good question, thinking about it more, since this issue was inspired by [SnippetLab Alfred Workflow](https://www.renfei.org/snippets-lab/manual/mac/tips-and-tricks/alfred-integration.html), I guess a good approach is not to dump all snippets but to ask the user to filter.

SnippetLabs ships with an Alfred Workflow that packages a binary that lets you query its database, but it doesn't appear to let you dump all of the snippets:

```
unzip /Applications/SnippetsLab.app/Contents/SharedSupport/Integrations/SnippetsLabAlfred.alfredworkflow -d /tmp/
/tmp/SnippetsLabAlfredWorkflow --action=search --query="OpenStack"
```
This will return a plist (XML formatted) of  Snippets IDs tagged with `#OpenStack`.

For `snipkit', I think a similar behavior makes sense.

**Comment by lemoony** [comment]
At long last, the next release will contain a new sub-command to export all snippets as JSON (implemented in https://github.com/lemoony/snipkit/pull/258 ). This goes hand-in-hand with feature request https://github.com/lemoony/snipkit/issues/233.

**Comment by lemoony** [comment]
Released with v1.5.0 👍 

There's also an example of how to customize snipkit for alternative workflows:

https://lemoony.github.io/snipkit/v1.5.0/getting-started/fzf/
